src/Avalonia.Base/Utilities/ValueSingleOrList.cs

```diff
@@ -81,7 +81,7 @@ public bool Remove(T value)
 
             if (EqualityComparer<T>.Default.Equals(Single, value))
             {
-                Single = default;
+                Single = default!;
```

Comment:
Shouldn't Single be nullable?

Reply:
At least with MaybeNull attribute.

Reply:
Yeah that's a tricky one. I kept `Single` as `T` because when it's in the valid state, it's guaranteed to be `T`. Ideally I think we'd throw an exception when reading `Single` if `IsSingle == false` in which case `T` is the correct return value for it.